docs/core/entity/fan.md

```diff
@@ -66,13 +75,32 @@ class FanEntity(ToggleEntity):
 class FanEntity(ToggleEntity):
     # Implement one of these methods.
 
-    def turn_on(self, speed: Optional[str] = None, **kwargs: Any) -> None:
+    def turn_on(self, speed: Optional[str] = None percentage: Optional[int] = None, **kwargs: Any) -> None:
         """Turn on the fan."""
 
-    async def async_turn_on(self, speed: Optional[str] = None, **kwargs: Any) -> None:
+    async def async_turn_on(self, speed: Optional[str] = None, percentage: Optional[int] = None, **kwargs: Any) -> None:
         """Turn on the fan."""
 ```
 
+:::tip `speed` is deprecated.
+
+For intergrations that implemented `speed` before the model changed to percentage,
+add the following code to the beginning of the function for backwards compatibility:
+
+```python
+        if percentage is not None and speed is None:
+            speed = self.percentage_to_speed(percentage)
+```
+
+For intergrations that implemented `percentage` after the model deprecated `speed`,
+add the following code to the beginning of the function for backwards compatibility:
```

Comment:
What will our transition plan be time wise? How long do we need to support the legacy mode when adding new fan platforms?

Reply:
I put a timeline in the document. Let me know if you think we should adjust it